wrapper/CSharp/wolfTPM.cs

```diff
@@ -0,0 +1,837 @@
+                                             pub_buffer, pub_buffer.Length, exp,
+                                             priv_buffer, priv_buffer.Length);
+            if (ret != TPM_RC_SUCCESS) {
+                Console.WriteLine("wolfTPM2_LoadRsaPrivateKey() failed.");
+            }
+        }
+
+        if (priv_key != IntPtr.Zero) {
+            if (wolfTPM2_CleanupKey(priv_key) != TPM_RC_SUCCESS) {
+                Console.WriteLine("wolfTPM2_CleanupKey(priv_key) failed.");
+            }
+        }
+
+        if (parent_key != IntPtr.Zero) {
+            if (wolfTPM2_CleanupKey(parent_key) != TPM_RC_SUCCESS) {
+                Console.WriteLine("wolfTPM2_CleanupKey(parent_key) failed.");
+            }
+        }
+
+        if (dev != IntPtr.Zero) {
+            if (wolfTPM2_SimpleCleanup(dev) != TPM_RC_SUCCESS) {
+                Console.WriteLine("wolfTPM2_SimpleCleanup() failed.");
+            }
+        }
+
+        Console.WriteLine("Leaving TryLoadRSAPrivateKey() with ret=" + ret);
+        Console.WriteLine("");
+    }
+
+    public static void TryImportRSAPrivateKey() {
+        int ret = TPM_RC_SUCCESS;
+        IntPtr dev = IntPtr.Zero;
+        IntPtr parent_key = IntPtr.Zero;
+        IntPtr priv_key = IntPtr.Zero;
+        IntPtr blob = IntPtr.Zero;
+        byte[] pub_buffer;
+        byte[] priv_buffer;
+        int exp = 0x10001;
+
+        pub_buffer = File.ReadAllBytes("rsa_public.bin");
+        priv_buffer = File.ReadAllBytes("rsa_private.bin");
+        PrintByteArray(pub_buffer);
+        PrintByteArray(priv_buffer);
+
+        dev = wolfTPM2_SimpleInit();
+        if (dev == IntPtr.Zero) {
+            Console.WriteLine("wolfTPM2_SimpleInit() failed.");
+            ret = -1;
+        }
+
+        if (ret == TPM_RC_SUCCESS) {
+            priv_key = wolfTPM2_GetEmptyKey();
+            if (priv_key == IntPtr.Zero) {
+                Console.WriteLine("wolfTPM2_GetEmptyKey() failed.");
+                ret = -1;
+            }
+        }
+
+        if (ret == TPM_RC_SUCCESS) {
+            parent_key = wolfTPM2_GetEmptyKey();
+            if (parent_key  == IntPtr.Zero) {
+                Console.WriteLine("wolfTPM2_GetEmptyKey() failed.");
+                ret = -1;
+            }
+        }
+
+        if (ret == TPM_RC_SUCCESS) {
+            blob = wolfTPM2_GetEmptyKeyBlob();
+            if (blob == IntPtr.Zero) {
+                Console.WriteLine("wolfTPM2_GetEmptyKeyBlob() failed.");
+                ret = -1;
+            }
+        }
+
+        if (ret == TPM_RC_SUCCESS) {
+            ret = wolfTPM2_ReadPublicKey(dev, parent_key, TPM2_PARENT_HANDLE);
+            if (ret != TPM_RC_SUCCESS) {
+                Console.WriteLine("wolfTPM2_ReadPublicKey() failed.");
+                ret = -1;
+            }
+        }
+
+        if (ret == TPM_RC_SUCCESS) {
+            ret = wolfTPM2_SetKeyAuthPassword(parent_key,
+                                              "ThisIsMyStorageKeyAuth", 22);
```

Comment:
It is better to match to pass the password bytes explicity.
```
string passwd = "ThisIsMYStorageKeyAuth"
byte[] pwbytes = Encoding.ASCII.GetBytes(passwd)
```
....

Reply:
Yes. This looks like a better approach. Will make the change.